src/main/kotlin/org/rust/ide/docs/RsQualifiedNameProvider.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.docs
+
+import com.intellij.ide.actions.QualifiedNameProvider
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.ext.RsQualifiedNamedElement
+import org.rust.lang.core.psi.ext.qualifiedName
+
+class RsQualifiedNameProvider : QualifiedNameProvider {
```

Comment:
I think we need to make custom implementation for `insertQualifiedName` as well because even now (without a suggestion to use `RsQualifiedName`), the combination of `Copy Reference` and `Paste` actions produces invalid reference sometimes, for example

```rust
//- main.rs of `foo` package
struct Foo; // copy reference to this struct

fn main() {
    foo::Foo; // result of paste action. Can be resolved
}
```